Titanium.Web.Proxy/Http/HttpWebClient.cs

```diff
@@ -208,5 +208,13 @@ internal void SetConnection(TcpConnection connection)
             //Read the response headers in to unique and non-unique header collections
             await HeaderParser.ReadHeaders(ServerConnection.StreamReader, Response.NonUniqueResponseHeaders, Response.ResponseHeaders);
         }
+
+        /// <summary>
+        /// Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.
+        /// </summary>
+        public void Dispose()
+        {
+            ServerConnection.Dispose();

```

Comment:
We can't call this here. Since we reuse this for same client.

So basically we create new HttpWebClient (per session), but we will reuse the server connection.